--- src/magics.h.orig	2021-06-08 16:45:53.152506180 +0200
+++ src/magics.h	2021-06-08 16:46:42.260581784 +0200
@@ -357,18 +357,6 @@
 inline MAGICS_NO_EXPORT string buildSharePath(const string& config, const string& aux = "") {
     string magplushome = getEnvVariable("MAGPLUS_HOME");
     if(magplushome.empty()) {
-#if defined(magics_HAVE_DLFCN_H)
-       Dl_info info;
-       if(dladdr((void*)getEnvVariable, &info)){
-            string libpath(info.dli_fname);
-            // remove libname and lib folder from path name
-            std::size_t found = libpath.find_last_of("/\\");
-            libpath = libpath.substr(0,found);
-            found = libpath.find_last_of("/\\");
-            magplushome = libpath.substr(0,found);
-       }
-       else 
-#endif
          magplushome = string(MAGICS_INSTALL_PATH);  
     }
     ostringstream out;
